include/libdnf/rpm/reldep.hpp

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright (C) 2018-2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_RPM_RELDEP_HPP
+#define LIBDNF_RPM_RELDEP_HPP
+
+#include "sack.hpp"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+
+namespace libdnf::rpm {
+
+/// @replaces libdnf/dnf-reldep.h:struct:DnfReldep
+/// @replaces libdnf/repo/solvable/Dependency.hpp:struct:Dependency
+/// @replaces hawkey:hawkey/__init__.py:class:Reldep
+class Reldep
+{
+public:
+    enum class ComparisonType {
+        NONE = 0,
+        GT = 1,
+        EQ = 2,
+        GT_EQ = 3,
+        LT = 4,
+        LT_EQ = 6
```

Comment:
We could either keep these as sequential values or use a bit mask, see:
https://github.com/rpm-software-management/libdnf/pull/930/files#diff-6377fb8e92bc3165e82f0b11978b18dfR32

Also rename `GT_EQ` to `GTE` and `LT_EQ` to `LTE` for consistency with the PR above.

Reply:
We can do that but I used identical values to libsolv in `pool.h`. See also`static_assert(static_cast<int>(Reldep::ComparisonType::EQ) == REL_EQ,
                  "Reldep::ComparisonType::EQ is not identical to solv/REL_EQ");` in `reldep.cpp`.

Reply:
For me, libdnf internal consistency is more important than consistency between libdnf and libsolv.

Reply:
I tried to refactor it according to request.